hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -3095,12 +3095,12 @@ LocatedBlock getAdditionalDatanode(String src, long fileId,
     final byte storagePolicyID;
     final List<DatanodeStorageInfo> chosen;
     final BlockType blockType;
-    checkOperation(OperationCategory.READ);
+    checkOperation(OperationCategory.WRITE);
     final FSPermissionChecker pc = getPermissionChecker();
     FSPermissionChecker.setOperationType(null);
     readLock();
     try {
-      checkOperation(OperationCategory.READ);
+      checkOperation(OperationCategory.WRITE);
```

Comment:
can we add comments here explaining why we check `WRITE` category even though we only obtain the `READ` lock?

Reply:
Sir, sorry for the late update. Please help me review it again, thanks.